processor/src/main/java/org/mapstruct/ap/internal/model/source/ForgedMethod.java

```diff
@@ -225,7 +225,7 @@ public ForgedMethodHistory getHistory() {
     }
 
     public boolean isAutoMapping() {
-        return mappingOptions.getValueMappings().isEmpty();
+        return MappingOptions.empty().equals( mappingOptions );
```

Comment:
Well `mappingOptions.getValueMappings()` is a list of `@ValueMapping` which is used enums. However, even though the modified check is more "correct" it is still not correct, because we create non-automapping `ForgedMethod` with `null` `MappingOptions`. I think that we need some explicit parameter or a new `AutoMappingForgedMethod` that will explicitly show that it was created due to automapping.